### ISSUE TITLE
Paginate does only work for markdown despite its syntax suggest to be html syntax

### ISSUE THREAD

**Issue by schlichtanders** [tlienart/JuDoc.jl#908]
Looking through the source code I realized that the pagination feature is an exception to the rule that curly braces ``{{ ... }}`` are used for html stuff.

Unlike this convention, the processing of pagination only happens for markdownfiles
https://github.com/tlienart/Franklin.jl/blob/master/src/manager/write_page.jl#L106-L150

**Comment by tlienart** [comment]
Sorry I missed your response here, could you give a small example of what you expected would work in `foot.html`? that will help illustrate the issue and either document it or fix it in next version, thanks

**Comment by schlichtanders** [comment]
sure. Let's say we have defined in config.md `mylist = ["1", "2", "3", ..., "10"]` then I having in foot.html
```
{{ paginate mylist 5 }}
```
I would expect this to output the different lists on the different subpages
```
* 1
* 2
* 3
* 4
* 5
```
and so on

however instead I the output is everywhere
```
%##PAGINATE##%
```
which is because PAGINATE is only parsed for markdown files and not html files currently (despite the {{ syntax suggests it to be also valid for html)

**Comment by tlienart** [comment]
ok yes, thanks for that. I understand that there's an ambiguity in that one might expect `{{...}}` to work anywhere. 

The thing though is if you put a `{{...}}` in a layout file (like `foot.html`) what does that mean; would you expect every single page of your blog get paginated? Another thing is that at the moment you can only have one "paginator" per page, which is kind of what constrained the "only in MD" thing. 

An alternative could be to have a separate syntax for the pagination like `<<paginate mylist 5>>` or `\paginate{mylist}{5}`. 

I'm tempted to leave things as they are but just making it abundantly clear what this command does and where it can be used but if you have suggestions for how it _could_ be made better or clearer, I'd be grateful

**Comment by schlichtanders** [comment]
Thank you for the explanation.
Actually the pagination triggers somehow and indeed generates multiple pages, also if I put it into the foot.html where it just doesn't display well. So in principle I can imagine this to be useful also in footer, i.e. for some generic paging in the footer.

The global restriction of having only one pagination applies also within a single MD file too, or?
I understand that it is a tough problem about how to combine multiple paginations on one page.

In total would go for `\paginate{mylist}{5}` as of now. (you can also leave the other {{ syntax for backwards compatibility, just without advertising it anywhere)

**Comment by tlienart** [comment]
> The global restriction of having only one pagination applies also within a single MD file too, or?
> I understand that it is a tough problem about how to combine multiple paginations on one page.

yes exactly; I'd be happy to make it more sophisticated but there would need to be a clear use case and an idea for how to manage it

> In total would go for `\paginate{mylist}{5}` as of now. (you can also leave the other {{ syntax for backwards compatibility, just without advertising it anywhere)

:+1:
